pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -350,12 +347,6 @@ public void healthCheck(@Suspended AsyncResponse asyncResponse,
                                 clientAppId(), topicName);
                         throw new RestException(Status.NOT_FOUND, "Topic [{}] not found after create.");
                     }
-                    Topic topic = topicOptional.get();
-                    // clean all subscriptions
-                    return FutureUtil.waitForAll(topic.getSubscriptions().values()
-                            .stream().map(Subscription::deleteForcefully).collect(Collectors.toList()))
```

Comment:
I am wondering if it is risky to remove this whole block. Instead, what if we simply called `Subscription::delete`? That would leave any subscription with active consumers in places. I think that should prevent the race I saw in the issue.

Additionally, we'd need to ignore the exceptionally completed future that the `delete` method returns.

Reply:
Moving this logic to the end of the `produce-consumer`.

Reply:
Hi, @michaeljmarshall 
If we use the ``delete`` method instead of ``deleteForce``. When the reader call ``closeAsync``method throws an exception and the connection is not closed. They will always be there and cause issues like #7724 mentioned.
So, I think we need to recheck the consumer last connected time to force deleted after the "delete" method to ensure this.